src/main/java/teammates/common/datatransfer/questions/FeedbackRubricQuestionDetails.java

```diff
@@ -908,6 +1036,117 @@ public void setNumOfRubricSubQuestions(int numOfRubricSubQuestions) {
+                    ++numOfResponsesPerSubQuestionPerChoice[i][choice];
+                    totalPerSubQuestion[i] += getRubricWeights().get(choice);
+                }
+            }
+        }
+
+        public String getHtmlForSubQuestion(int subQuestion) {
+            StringBuilder html = new StringBuilder(100);
+            String alphabeticalIndex = StringHelper.integerToLowerCaseAlphabeticalIndex(subQuestion + 1);
+            String subQuestionString = SanitizationHelper.sanitizeForHtml(alphabeticalIndex + ") "
+                    + getRubricSubQuestions().get(subQuestion));
+            DecimalFormat df = new DecimalFormat("#");
+            DecimalFormat dfAverage = new DecimalFormat("0.00");
+
+            List<String> cols = new ArrayList<>();
+
+            cols.add(recipientTeam);
+            cols.add(recipientName);
+            cols.add(subQuestionString);
+
+            for (int i = 0; i < getNumOfRubricChoices(); i++) {
+                cols.add(Integer.toString(numOfResponsesPerSubQuestionPerChoice[subQuestion][i]));
+            }
+
+            cols.add(df.format(totalPerSubQuestion[subQuestion]));
+            cols.add(dfAverage.format(totalPerSubQuestion[subQuestion] / getNumOfRubricSubQuestions()));
+
+            for (String col : cols) {
+                html.append(
+                        Templates.populateTemplate(FormTemplates.RUBRIC_RESULT_RECIPIENT_STATS_BODY_ROW_FRAGMENT,
+                        Slots.RUBRIC_RECIPIENT_STAT_CELL, col));
+            }
+
+            return html.toString();
+        }
+
+        public String getHtmlForAllSubQuestions() {
+            StringBuilder html = new StringBuilder(100);
+
+            for (int i = 0; i < getNumOfRubricSubQuestions(); i++) {
+                String subQuestionStats = getHtmlForSubQuestion(i);
+                html.append(Templates.populateTemplate(
+                        FormTemplates.RUBRIC_RESULT_RECIPIENT_STATS_BODY_FRAGMENT,
+                        Slots.RUBRIC_RECIPIENT_STAT_ROW, subQuestionStats));
+            }
+
+            return html.toString();
+        }
+
+        public String getCsvForSubQuestion(int subQuestion) {
+            StringBuilder csv = new StringBuilder(100);
+            String alphabeticalIndex = StringHelper.integerToLowerCaseAlphabeticalIndex(subQuestion + 1);
+            String subQuestionString = SanitizationHelper.sanitizeForCsv(alphabeticalIndex + ") "
+                    + getRubricSubQuestions().get(subQuestion));
+            DecimalFormat df = new DecimalFormat("#");
+            DecimalFormat dfAverage = new DecimalFormat("0.00");
+
+            csv.append(recipientTeam).append(',').append(recipientName).append(',').append(recipient)
+                    .append(',').append(subQuestionString);
+
+            for (int i = 0; i < getNumOfRubricChoices(); i++) {
+                csv.append(',').append(Integer.toString(numOfResponsesPerSubQuestionPerChoice[subQuestion][i]));
```

Comment:
In this code and in similar code that constructs a single row (header or body, HTML or CSV), there are 3 distinct chunks:
1. Recipient identification (team, name, email, subquestion)
1. The recipient data itself
1. Aggregates of the reicipient data (total, average)

It would probably be a good idea to extract these 3 chunks into methods, or at least mark them out clearly with comments, so that the code isn't so dense.